compiler/rustc_interface/src/passes.rs

```diff
@@ -973,7 +979,7 @@ fn analysis(tcx: TyCtxt<'_>, (): ()) -> Result<()> {
     // lint warnings and so on -- kindck used to do this abort, but
     // kindck is gone now). -nmatsakis
     if sess.has_errors() {
-        return Err(ErrorReported);
+        return Err(ErrorReported::new_i_know_what_im_doing());
```

Comment:
`sess.has_errors()` should be able to provide a `ErrorReported` - that might be the most convenient way to get it in many places.